Experiment 1

Aim:- Data wrangling with numpy pandas.


LO:- LO1

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("iris(1).csv")
df.head()

sepal_length  sepal_width  petal_length  petal_width species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal_length  150 non-null    float64
 1   sepal_width   150 non-null    float64
 2   petal_length  150 non-null    float64
 3   petal_width   150 non-null    float64
 4   species       150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


In [4]:
df.duplicated().sum()
df.isnull().sum()

sepal_length    0
sepal_width     0
petal_length    0
petal_width     0
species         0
dtype: int64

1. What are key features (columns) in the dataset?
-> sepal_length, sepal_width, petal_length, petal_width, species

2. What is the data type of each feature?
-> sepal_length, sepal_width, petal_length, petal_width: float ; species: object

3. Are there any missing values? which features have most missing data?
-> No missing values in the dataset.

4. What is size and shape of the dataset?
-> size: 150 ; shape: (150, 5)

5. Are there any duplicate rows in the dataset?
-> No duplicate rows in the dataset.

1. What are the possible reasons for missing values in a dataset?
-> Data entry errors, equipment malfunction, Data Integration Issues, Deletion of Data, Filtering or Sampling Issues are some possible reasons for missing values in a dataset.

2. How should missing data be handled for numeric features?
-> For numeric features, missing data can be handled by imputation methods such as mean, median, mode, KNN, or regression.

3. How should missing data be handled for categorical features?
-> For categorical features, missing data can be handled by directly removing rowa or columns, imputation methods such as mode, or encoding methods such as one-hot encoding or label encoding.

Conclusion:- This experiment shows basic data wrangling and basic understanding of ways to find info of dataset 

PO's achieved:- PO1, PO2, PO3, PO4, PO5